src/targeting.js

```diff
@@ -173,19 +167,25 @@ function getAlwaysUseBidTargeting(adUnitCodes) {
 
 function getBidLandscapeTargeting(adUnitCodes) {
   const standardKeys = CONSTANTS.TARGETING_KEYS.concat(NATIVE_TARGETING_KEYS);
-
-  return $$PREBID_GLOBAL$$._bidsReceived
-    .filter(adUnitsFilter.bind(this, adUnitCodes))
-    .map(bid => {
-      if (bid.adserverTargeting) {
-        return {
-          [bid.adUnitCode]: getTargetingMap(bid, standardKeys.filter(
-            key => typeof bid.adserverTargeting[key] !== 'undefined') // mainly for possibly
-            // unset hb_deal
-          )
-        };
-      }
-    }).filter(bid => bid); // removes empty elements in array
+  const bids = [];
+  // bucket by adUnitcode
+  let buckets = groupBy($$PREBID_GLOBAL$$._bidsReceived, 'adUnitCode');
+  // filter top bid for each bucket by bidder
+  Object.keys(buckets).forEach(bucketKey => {
+    let bidsByBidder = groupBy(buckets[bucketKey], 'bidderCode');
+    Object.keys(bidsByBidder).forEach(key => bids.push(bidsByBidder[key].reduce(getHighestCpm, getEmptyBid())));
+  });
+  // populate targeting keys for the remaining bids
+  return bids.map(bid => {
+    if (bid.adserverTargeting) {
+      return {
+        [bid.adUnitCode]: getTargetingMap(bid, standardKeys.filter(
+          key => typeof bid.adserverTargeting[key] !== 'undefined') // mainly for possibly
+          // unset hb_deal
```

Comment:
Yeah, although it's not stale. It's saying that `hb_deal` isn't always defined even though it's in the default key set. I think it should be obvious from the code though.